model/connect/Database.php

```diff
@@ -630,18 +641,24 @@ public function databaseList() {
 	 * @return boolean Flag indicating success
 	 */
 	public function selectDatabase($name, $create = false, $errorLevel = E_USER_ERROR) {
-		if (!$this->schemaManager->databaseExists($name)) {
-			// Check DB creation permisson
-			if (!$create) {
-				if ($errorLevel !== false) {
-					user_error("Attempted to connect to non-existing database \"$name\"", $errorLevel);
-				}
-				// Unselect database
-				$this->connector->unloadDatabase();
-				return false;
+		// In case our live environment is locked down, we can bypass a SHOW DATABASE check
+		$canConnect =
+			(Director::isLive() && Config::inst()->get(get_class($this), 'optimistic_live_connect'))
```

Comment:
I think so, otherwise we're just likely to see unexpected behaviour on live releases that were not seen on test sites (or problems on test sites that can't resolve)